File: java/spring-boot-demo/src/main/java/com/greatbit/xgn/console/domain/LoginUser.java
package com.greatbit.xgn.console.domain;

import javax.persistence.*;
import java.io.Serializable;
import java.sql.Timestamp;
import java.util.HashSet;
import java.util.Set;

@Entity
@Table(name = "gn_user")
public class LoginUser implements Serializable {
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private long id;
    @Column(length = 50, unique = true)
    private String username;
    @Column(length = 50)
    private String password;
    @Column
    private Timestamp loginDate = new Timestamp(System.currentTimeMillis());
    @Column(length = 20)
    private String loginIp;

    @ManyToMany(fetch = FetchType.EAGER)
    @JoinTable(name = "gn_user_role", joinColumns = {@JoinColumn(name = "u_id", referencedColumnName = "id")}, inverseJoinColumns = {@JoinColumn(name = "t_id", referencedColumnName = "id")})
    private Set<Role> roles = new HashSet<Role>();

    public String getUsername() {
        return username;
    }

    public void setUsername(String username) {
        this.username = username;
    }

    public String getPassword() {
        return password;
    }

    public void setPassword(String password) {
        this.password = password;
    }

    public Timestamp getLoginDate() {
        return loginDate;
    }

    public void setLoginDate(Timestamp loginDate) {
        this.loginDate = loginDate;
    }

    public String getLoginIp() {
        return loginIp;
    }

    public void setLoginIp(String loginIp) {
        this.loginIp = loginIp;
    }

    public Set<Role> getRoles() {
        return roles;
    }

    public void setRoles(Set<Role> roles) {
        this.roles = roles;
    }
}

File: scala/finatra-demo/src/main/scala/com/example/jpa/LoginUser.scala
package com.example.jpa

import javax.persistence._

import scala.beans.BeanProperty
import scala.collection.mutable.ListBuffer

@Entity
@Table(name = "t_user")
class LoginUser extends BaseEntity {
  @BeanProperty
  @Column(nullable = false, unique = true)
  var username: String = _

  @BeanProperty
  @Column
  var password: String = _

  @BeanProperty
  @Column
  var age: Int = _

  @BeanProperty
  @Column
  var address: String = _

  @BeanProperty
  @ManyToMany(cascade = Array(CascadeType.MERGE), fetch = FetchType.EAGER)
  @JoinTable(name = "user_role", joinColumns = Array(new JoinColumn(name = "user_id", referencedColumnName = "id")),
    inverseJoinColumns = Array(new JoinColumn(name = "role_id", referencedColumnName = "id")))
  var roles: ListBuffer[Role] = _
}

object LoginUser {
  def apply(username: String, password: String, age: Int, address: String): LoginUser = {
    val user = new LoginUser
    user.username = username
    user.password = password
    user.age = age
    user.address = address
    user
  }
}